src/platform/input/win32mouse.cpp

```diff
@@ -0,0 +1,238 @@
+/**
+ * @file
+ *
+ * @author DuncansPumpkin
+ *
+ * @brief Win32 Mouse Input interface.
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+
+#include "win32mouse.h"
+#include "editmain.h"
+#include "gameclient.h"
+
+#ifdef PLATFORM_WINDOWS
+#include <windowsx.h>
+#include <winuser.h>
+#endif
+
+#ifdef PLATFORM_WINDOWS
+// 0x00A31F00 (note first entry not used as it is CURSOR_NONE) 8 directions
+static HCURSOR g_LoadedCursors[CURSOR_COUNT][8];
+#endif
+
+// 0x0073C0A0
+Win32Mouse::Win32Mouse() :
+    m_nextFreeIndex(0), m_nextGetIndex(0), m_currentWin32Cursor(0), m_cursorDirection(0), m_unknownbool(false)
+{
+    std::memset(m_eventBuffer, 0, sizeof(m_eventBuffer));
+#ifdef PLATFORM_WINDOWS
+    std::memset(g_LoadedCursors, 0, sizeof(g_LoadedCursors));
+#endif
+}
+
+// 0x0073C130
+void Win32Mouse::Init()
+{
+    Mouse::Init();
+    m_inputMovesAbsolute = 1;
+}
+
+static void Load_Cursor(const char *file_location, uint32_t load_index, uint32_t load_direction)
+{
+#ifdef PLATFORM_WINDOWS
+    if (g_LoadedCursors[load_index][load_direction] != NULL) {
+        // Cursor at this position already loaded. Ignoring subsequent load.
+        return;
+    }
+    g_LoadedCursors[load_index][load_direction] = LoadCursorFromFile(file_location);
+#endif
+}
+
+// 0x0073C260
+void Win32Mouse::Init_Cursor_Resources()
+{
+    // First cursor is not loaded. As this is CURSOR_NONE
+    for (auto i = 1; i < CURSOR_COUNT; ++i) {
+        auto &cursor_info = m_cursorInfo[i];
+        if (cursor_info.texture_name.Is_Empty()) {
+            continue;
+        }
+
+        if (cursor_info.directions <= 1) {
+            char buffer[256]{};
+            snprintf(buffer, 256, "data\\cursors\\%s.ANI", cursor_info.texture_name.Str());
```

Comment:
While I realise that the original binary probably has the strings like this, it would probably be best for future cross compatibility to use a correctly case sensitive path based on the installed data. "Data/Cursors/%s.ani" to be specific. That assumes that the winapi function handles the posix path separators correctly too, but most winapi functions do. Thoughts?

Reply:
Yes that makes sense. I've made that change. TBH should probably use the filesystem library to handle all of this eventually.